src/services/paras/ParasService.ts

```diff
@@ -0,0 +1,441 @@
+import { u32 } from '@polkadot/types';
+import { Option, Vec } from '@polkadot/types/codec';
+import {
+	AccountId,
+	BalanceOf,
+	BlockHash,
+	BlockNumber,
+	FundInfo,
+	ParaGenesisArgs,
+	ParaId,
+	ParaLifecycle,
+	WinningData,
+} from '@polkadot/types/interfaces';
+import { ITuple } from '@polkadot/types/types';
+import BN from 'bn.js';
+import { InternalServerError } from 'http-errors';
+
+import {
+	AuctionPhase,
+	IAuctionsCurrent,
+	ICrowdloans,
+	ICrowdloansInfo,
+	IFund,
+	ILeaseInfo,
+	ILeasesCurrent,
+	IParas,
+	LeaseFormatted,
+	ParaType,
+} from '../../types/responses';
+import { IOption, isSome } from '../../types/util';
+import { AbstractService } from '../AbstractService';
+
+// This was the orgiginal value in the rococo test net. Once the exposed metadata
+// consts makes its way into `rococo-v1` this can be taken out.
+const LEASE_PERIODS_PER_SLOT_FALLBACK = 4;
+
+export class ParasService extends AbstractService {
+	/**
+	 * Get crowdloan information for a `paraId`.
+	 *
+	 * @param hash `BlockHash` to make call at
+	 * @param paraId ID of para to get crowdloan info for
```

Comment:
It might not necessarily be a parachain because it could be a parathread. So I chose to go with the convention of just reffering to both as "para". Open to other ideas, but as far as I can tell that is the direction the polkadot source code is heading.

Reply:
Ah ofc, you're right.